src/LCRmeter-Metrohm_AutolabPGSTAT/main.py

```diff
@@ -0,0 +1,339 @@
+    def disconnect(self) -> None:
+        """Disconnect from the Metrohm Autolab LCRmeter."""
+        self.autolab.Disconnect()
+
+    def initialize(self) -> None:
+        """Initialize the Metrohm Autolab LCRmeter."""
+
+    def deinitialize(self) -> None:
+        """Reset device and close connection."""
+
+    def configure(self) -> None:
+        """Set bias and measurement parameters with start values from GUI."""
+        # Initial Measurement Parameters
+        self.set_frequency(self.frequency)
+        self.set_dc_bias(self.dc_bias)
+        self.set_ac_voltage(self.ac_amplitude)
+
+        # General Measurement Parameters
+        self.Fra.WaveType = self.Fra.WaveType.Sine
+
+        # If needed, this parameter could be added to the GUI
+        self.Fra.MinimumIntegrationTime = 0.125
+
+        self.Fra.MinimumIntegrationCycles = self.number_of_cycles
+        self.set_measure_range(self.measure_range)
+        self.set_speed(self.speed)
+
+        # Prepare device
+        self.Ei.EnableDsgInput = True
+        self.Ei.CellOnOff = self.Ei.EICellOnOff.On
+
+        InstrumentExtensions.SwitchFraOn(self.autolab)
+        time.sleep(1)  # wait to stabilize
+
+    def unconfigure(self) -> None:
+        """Reset device."""
+        self.Ei.CellOnOff = self.Ei.EICellOnOff.Off
+        self.Ei.EnableDsgInput = False
+
+    def apply(self) -> None:
+        """Apply settings."""
+        if self.sweepmode != "None":
+            sweep_value = float(self.value)
+            self.handle_set_value(self.sweepmode, sweep_value)
+
+        if self.stepmode != "None":
+            step_value = float(self.stepvalue)
+            self.handle_set_value(self.stepmode, step_value)
+
+    def measure(self) -> None:
+        """Retrieve Impedance results from device."""
+        # Run the measurement
+        self.Fra.Start()
+
+        # Retrieve measured values
+        self.resistance, self.reactance = self.measure_resistance_reactance()
+
+        # Currently, the set values are returned
+        # It is unclear whether it is possible to measure the applied values for frequency and bias voltage
+        self.measured_frequency = self.measure_frequency()
+        self.measured_dc_bias = self.measure_dc_bias()
+
+    def call(self) -> list[float]:
+        """Return ["R", "X", "Frequency", "Voltage bias" or "Voltage level"]."""
+        return [self.resistance, self.reactance, self.measured_frequency, self.measured_dc_bias]
+
+    """ here wrapped functions start """
+
+    def get_adx_path(self) -> None:
+        """Read the ini file and set the adx path."""
+        try:
+            ini_data = self.get_configoptions("Autolab HardwareSetup")
+            adk_path = ini_data["adk_path"]
+        except KeyError:
+            adk_path = ""
+            debug("No adk_path found in ini file.")
+
+        if Path(adk_path).exists() is False:
+            msg = "Invalid adk_path. Ensure the LCRmeter_Metrohm-AutolabPGSTAT.ini file is set up correctly."
```

Comment:
So if there is no adk_path I first get the debug message that adk_path is not found in ini file, and afterwards I get the exception that the adk_path in invalid. Not really a big issue, just wanted to point it out.